### DISCUSSION TITLE
How to run "Proxmox VE Processor Microcode" script properly?

### DISCUSSION THREAD

**Discussion by kevindd992002** [tteck/Proxmox#3077] [Help Needed]
So I ran this script in the shell and chose `intel-microcode_3.20240514.1_amd64.deb`. The script completed successfully and asked me to reboot pve to apply the changes, which I did. To validate, I ran the script again and it still says `Current microcode revision:Not found`. I also checked with `dmesg | grep "microcode updated early to"` and there were not results. So then I tried the same with the other microcode versions and got the exact same results.

Thoughts?

**Comment by klingon888** [comment]
Interested to know answer too as I have the same result as OP with Intel N100 and N5105, latest Proxmox versions.

Running: journalctl -k | grep -iE "microcode" shows the following:
Jun 01 09:58:08 n100 kernel: Register File Data Sampling: Vulnerable: No microcode
Jun 01 09:58:08 n100 kernel: microcode: Current revision: 0x0000000e

**Comment by tteck** [comment]
fixed view current microcode version https://github.com/tteck/Proxmox/commit/cd1d295ae1e87d3c5b0f3bd87143167fe346bb3c

Not all CPU microcode patches are applicable to all CPU models. If your CPU wasn't part of the patch, then the microcode version won't change because it's not intended for your specific CPU.

**Reply by kevindd992002** [reply]
> fixed view current microcode version [cd1d295](https://github.com/tteck/Proxmox/commit/cd1d295ae1e87d3c5b0f3bd87143167fe346bb3c)
> 
> Not all CPU microcode patches are applicable to all CPU models. If your CPU wasn't part of the patch, then the microcode version won't change because it's not intended for your specific CPU.

So if all patches aren't applicable to my CPU, then doing a `dmesg | grep "microcode updated early to"` is expected to not output any results?

**Reply by kevindd992002** [reply]
Any updates here @tteck ?

**Comment by damjank12** [comment]
Since wee are talking about proper way to run update - I have a question - when I run this script, it gives me two results to select from what to use for update - no idea what to select, or better, what to proeprly select... I have Xeon Gold series, and I get this to choose from
<img width="571" alt="Screenshot 2024-07-26 at 09 26 47" src="https://github.com/user-attachments/assets/acbd1dd6-e13a-4f16-bf0e-948c082bcac4">
Any idea what to select or how to decide (not based on chance ofc.,m but I did not find any relevant information to help out).
Thank you!
